src/contracts/core/StrategyManager.sol

```diff
@@ -262,90 +265,93 @@ contract StrategyManager is
-                "StrategyManager._addShares: deposit would exceed MAX_STAKER_STRATEGY_LIST_LENGTH"
+                "StrategyManager._addScaledShares: deposit would exceed MAX_STAKER_STRATEGY_LIST_LENGTH"
             );
             stakerStrategyList[staker].push(strategy);
         }
 
-        // add the returned shares to their existing shares for this strategy
-        stakerStrategyShares[staker][strategy] += shares;
+        // add the returned scaled shares to their existing scaled shares for this strategy
+        stakerStrategyScaledShares[staker][strategy] += scaledShares;
 
-        emit Deposit(staker, token, strategy, shares);
+        emit Deposit(staker, token, strategy, scaledShares);
     }
 
     /**
      * @notice Internal function in which `amount` of ERC20 `token` is transferred from `msg.sender` to the Strategy-type contract
-     * `strategy`, with the resulting shares credited to `staker`.
-     * @param staker The address that will be credited with the new shares.
+     * `strategy`, with the resulting scaledShares credited to `staker`.
+     * @param staker The address that will be credited with the new scaledShares.
      * @param strategy The Strategy contract to deposit into.
      * @param token The ERC20 token to deposit.
      * @param amount The amount of `token` to deposit.
-     * @return shares The amount of *new* shares in `strategy` that have been credited to the `staker`.
+     * @return scaledShares The amount of *new* scaled shares in `strategy` that have been credited to the `staker`.
      */
     function _depositIntoStrategy(
         address staker,
         IStrategy strategy,
         IERC20 token,
         uint256 amount
-    ) internal onlyStrategiesWhitelistedForDeposit(strategy) returns (uint256 shares) {
+    ) internal onlyStrategiesWhitelistedForDeposit(strategy) returns (uint256 scaledShares) {
         // transfer tokens from the sender to the strategy
         token.safeTransferFrom(msg.sender, address(strategy), amount);
 
         // deposit the assets into the specified strategy and get the equivalent amount of shares in that strategy
-        shares = strategy.deposit(token, amount);
+        uint256 shares = strategy.deposit(token, amount);
 
-        // add the returned shares to the staker's existing shares for this strategy
-        _addShares(staker, token, strategy, shares);
+        // scale strategy shares before storing
+        scaledShares = delegation.getStakerScaledShares(staker, strategy, shares);
 
-        // Increase shares delegated to operator, if needed
-        delegation.increaseDelegatedShares(staker, strategy, shares);
+        // add the returned scaled shares to the staker's existing scaled shares for this strategy
+        _addScaledShares(staker, token, strategy, scaledShares);
 
-        return shares;
+        // Increase scaled shares delegated to operator, if needed
+        delegation.increaseDelegatedScaledShares(staker, strategy, scaledShares);
```

Comment:
wonder if a `increaseDelegatedShares` that scales and returns the scaled shares makes more sense than calling DM twice to scale and then increase?

Reply:
Definitely a good optimization, will add this later because I think its easier to view contracts diff as currently implemented.